src/components/main/files/browser/mod.rs

```diff
@@ -45,24 +46,76 @@ pub fn FileBrowser(cx: Scope<Props>) -> Element {
         div {
             id: "browser",
             (cx.props.show_new_folder).then(|| rsx!(
-                NewFolder {
-                    state: State::Primary
+                div {
+                    class: "item file",
+                    NewFolder {
+                        state: State::Primary
+                    }
                 }
             )),
             files_sorted.iter().filter(|item| item.item_type() == ItemType::FileItem).map(|file| {
+
                 let file_extension = std::path::Path::new(&file.name())
                 .extension()
                 .unwrap_or_else(|| std::ffi::OsStr::new(""))
                 .to_str()
                 .unwrap()
                 .to_string();
 
-                rsx!( File {
-                    name: file.name(),
-                    state: State::Secondary,
-                    kind: file_extension,
-                    size: file.size(),
-                })
+                rsx!(
+                    div {
+                        class: "dropdown", 
+                        div {
+                            class: "item file",
+                            File {
+                                name: file.name(),
+                                state: State::Secondary,
+                                kind: file_extension,
+                                size: file.size(),
+                                thumbnail: file.thumbnail(),
+                            }
+                        }
+                        div {
+                            class: "dropdown-content", 
+                            IconButton {
+                                icon: Shape::X,
+                                state: ui_kit::icon_button::State::Secondary,
+                                on_pressed: move |_| {
+                                    let file_storage = cx.props.storage.clone();
+                                    let file_name = file.name();
+                                    cx.spawn({
+                                        to_owned![file_storage, file_name];
+                                        async move {
+                                        let mut write_storage = file_storage.write();
```

Comment:
As we discussed over dm, you should do `file_storage.remove` directly without holding the write lock itself.